scarb/src/core/resolver.rs

```diff
@@ -38,4 +39,57 @@ impl Resolve {
             .iter(&self.graph)
             .unique()
     }
+
+    pub fn solution_with_target_kind(
+        &self,
+        root_package: PackageId,
+        target_kind: TargetKind,
+    ) -> Vec<PackageId> {
+        assert!(&self.graph.contains_node(root_package));
+        let filtered_graph = EdgeFiltered::from_fn(&self.graph, move |(_node_a, _node_b, edge)| {
+            edge.accepts_target_kind(target_kind.clone())
+        });
+        Dfs::new(&filtered_graph, root_package)
+            .iter(&filtered_graph)
+            .unique()
+            .collect_vec()
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct DependencyEdge(SmallVec<[TargetKind; 4]>);
```

Comment:
Why 4 instead of 1? Can't we represent wildcards here?

Reply:
Right now empty vec = wildcard, hence https://github.com/software-mansion/scarb/blob/ae88e56894898b59005a6f7de1361c1829df07a3/scarb/src/core/resolver.rs#L79